### DISCUSSION TITLE
Pretty pytest exceptions for threads

### DISCUSSION THREAD

**Discussion by jordanisaacs** [pytest-dev/pytest#12682] [Q&A]
I have tests that spawn threads which can raise exceptions causing the test to fail. Is there a way to pass in the nice pytest exception handler/formatter to those threads so I can get more then just a stack trace. E.g. I want to show the locals.

Right now, I just print the exception and then call  `assert False` on the main thread. This results in a nice pytest exception but in the wrong place. From what I can tell https://github.com/pytest-dev/pytest/blob/5a0158368798f466ac74c8e89aef40829cd86d5c/src/_pytest/_code/code.py#L618 is what I want but it is part of the private library.

This is what the high level code looks like at the moment:

```python
def test_threads():
    def run_async():
         try:
               do_things()
         except:
               # I want to do the pretty pytest exception formatting here
               LOG.exception("do things failed", exc_info=e)
    
    handle = run_in_thread(run_async)
    do_other_things()
    if handle.error():
           assert False
```

**Comment by RonnyPfannschmidt** [comment]
The Exception info class provides the formatting

**Reply by RonnyPfannschmidt** [reply]
But how about composing a exception group from futures instead of logging in threads
